proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,86 @@
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many products (Apps / Tools / Frameworks / Utilities / etc. ...) consist of multiple interdependent packages which should be published together as a single set. Currently, NuGet provides no functionality for definding / grouping packages into such sets. This results in package owners having to keep track of these package sets on their own. There are three specific sets of scenarios which this feature will simplify dramatically:
+1. When preparing a release, product owners would like to have a mechanism to pre-stage their product packages, so that the release will consist only of a single action of marking the set of packages as `publishged`, which would result all the packages in the set to be marked as published at the same time.
+2. Sometimes developers have to take a dependency on several related packages in their projects, which are part of the same "release". Without having a concept of the `release` developers have to guess which versions of those packages are compatible. As a result, most of hte time developers would simply pick the latest version of all these packages, which somtimes can lead to incompatibility issues down the road.
+3. Keeping a package healthy on NuGet also means keeping the list of packages aligned with it support policy. Hence, package owners try to deprecate packages which are out of support. Without ability to group packages, package owners end up keeping track of all the packages which belong to the same release and run the deprecation over a discrete set of packages when necessary. Being able to group packages together would enable deprecating a whole release at once, knowing that all the packages which belong to that release will get deprecated.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+There is a new field in the package metadata, which indicates the unique release of the product the package belongs to. This field is called `release` and the value is scoped to the publisher of the package.
+
+When a package is being built, that field in the metadata is set to the same value for all the related packages, that are built together.
+
+Now, with this information in place, new set of NuGet APIs / extensions allow the user to run the following actions:
+- **Search / Query for packages based on a release** for a given publisher
+- **Deprecate a relase** for current publisher
```

Comment:
Maybe my questions are more suitable for a "technical spec", rather than a "user spec", but this is how my brain works. I can't help but think about all the little details as if I was being asked to implement it immediately.

With the current proposal, where the package author decides the release name before creating their package, what happens if a package is pushed to a release after the release has already been marked as deprecated? Should the new package be immediately marked as deprecated as well? Should a release become "locked" when marked as deprecated, returning an error is any new package is uploaded with the same release?  Should it be server implementation dependent?

Reply:
I agree this is too technical. From the UX standpoint that doesn't make sense. If the release is deprecated I'd expect no further modifications to it should be made. So I'd say this should result in an error.